### PR TITLE
Implement RFC #1725 (read_unaligned, write_unaligned)

### PR DESCRIPTION
**Pull request by sfackler**
cc #37955

r? @alexcrichton

### CODE REVIEW

File: src/libcore/ptr.rs [hunk 1]
```diff
@@ -171,6 +213,46 @@ pub unsafe fn write<T>(dst: *mut T, src: T) {
...
+///     assert_eq!(std::ptr::read_unaligned(y), 12);
+/// }
+/// ```
+#[inline]
+#[unstable(feature = "ptr_unaligned", issue = "37955")]
+pub unsafe fn write_unaligned<T>(dst: *mut T, src: T) {
+    copy_nonoverlapping(&src as *const T as *const u8,
+                        dst as *mut u8,
+                        mem::size_of::<T>());
+}
```
Comment (bluss) [inline src/libcore/ptr.rs]: This consumes `src` and must forget `src` here. This calls for a test!

**Comment by alexcrichton** [comment]
Looks good to me, thanks @sfackler!

r=me with @bluss's comments and some tests.

File: src/libcore/ptr.rs (hunk 1 above)
Comment (sfackler) [inline src/libcore/ptr.rs]: Fixed and added a test.

Comment (bluss) [inline src/libcore/ptr.rs]: Great

**Pull request merged**